tsconfig.json

```diff
@@ -20,5 +20,5 @@
 		}
 	},
 	"include": ["lib", "types", "package.json"],
-	"exclude": ["**/*.test.js", "**/__tests__/"]
+	"exclude": ["**/*.test.js", "**/__tests__/", "lib/**/*.cjs"]
```

Comment:
off topic

Why would we ignore `"**/*.test.js"` exclusively?
i.e. why not mjs/cjs as well?

Reply:
Because many modifications are necessary to pass type-check for test files.

`"**/*.test.js"` is redundant. We can simplify the config. I'll open another PR to fix it.

Reply:
I've opened PR #7059 to support also `"**/*.test.mjs"`. And I reconsider it makes sense to keep `"**/*.test.js"`, for example, see `system-tests/001/*.test.mjs`.